docs/docs/building-an-ecommerce-site-with-shopify.md

```diff
@@ -0,0 +1,195 @@
+---
+title: Building an e-commerce site with Shopify
+---
+
+In this tutorial, you will setup a new Gatsby website that fetches product data from [Shopify](https://www.shopify.com). The site displays a list of all products on a product listing page, and a page for every product in the store.
+
+If you are already comfortable with Gatsby and Shopify, you might want to check out the [Gatsby Shopify starter](https://www.gatsbyjs.org/starters/AlexanderProd/gatsby-shopify-starter/), which provides many of the same features as this example.
+
+## Setting up your Shopify account
+
+1. Create a new [Shopify account](https://www.shopify.com) and store if you don't have one.
+2. Create a Private App in your store by navigating to **Apps**, then "Manage private apps"
+3. Create a new private app and enable the [Shopify Storefront API](https://help.shopify.com/en/api/storefront-api). Make sure to also grant access to read product and customer tags.
+
+## Set up the Gatsby Shopify plugin
+
+1. If you do not already have one ready, [create a Gatsby site](https://www.gatsbyjs.org/docs/quick-start).
+
+2. Install the `gatsby-source-shopify` plugin and `shopify-buy` package.
+
+```shell
+npm install --save gatsby-source-shopify shopify-buy
+```
+
+3. Enable and configure the plugin in your `gatsby-config.js` file.
+
+```javascript:title=/gatsby-config.js
+plugins: [
+  {
+    resolve: `gatsby-source-shopify`,
+    options: {
+      // The domain name of your Shopify shop.
+      shopName: `[some-shop].myshopify.com`,
+
+      // The storefront access token
+      accessToken: `[token]`,
+    },
+  },
+]
+```
```

Comment:
I would also add here a notice about private accestoken And link to the environment variable page?